internal/consensus/state.go

```diff
@@ -1638,6 +1639,13 @@ func (cs *State) defaultDoPrevote(ctx context.Context, height int64, round int32
 		}
 	}
 
+	// If ProposalBlock is still nil, prevote nil.
+	if cs.ProposalBlock == nil {
```

Comment:
This logic is in the TM repo too, but because we gossip tx keys, we should also check after the  above calls to see if it's still nil, if so just vote nil as the later parts depends on proposal block being an actual object https://github.com/tendermint/tendermint/blob/v0.37.0-rc1/consensus/state.go#L1279-L1284